### DISCUSSION TITLE
execute_rpc crash

### DISCUSSION THREAD

**Discussion by kangyite** [tdslite/tdslite#26] [Q&A]
When using `execute_rpc()`, sometimes it crashes and show `assert failed: tdsl::detail::tdsl_driver<NetImpl>::sql_command_rpc_result tdsl::detail::tdsl_driver<NetImpl>::execute_rpc(T, tdsl::span<tdsl::detail::sql_parameter_binding>, tdsl::detail::tdsl_driver`

When I check the function it seems like tds_ctx.is_authenticated() return 0 and assert terminate the program. Why is it not authenticated?  
and is it possible to reconnect the context to the server?

**Comment by xmkg** [comment]
Hello @kangyite 

> When I check the function it seems like tds_ctx.is_authenticated() return 0 and assert terminate the program. Why is it not authenticated?
> and is it possible to reconnect the context to the server?

The is_authenticated flag is set when the server sends LOGINACK packet as a response to the LOGIN packet, which indicates that the login attempt was successful. Are you checking the `driver.connect()`'s return value? Because the only reason I can think of why this happens is the driver is failing to authenticate against the server, but the sketch still proceeds to execute SQL commands without checking whether `connect` has succeeded or not.

**Reply by kangyite** [reply]
Thanks for the reply, I never get the 1st error after checking `driver.connect()`'s return value. But sometimes I get this error `assert failed: uint32_t tdsl::net::network_io_base<Implementation>::do_receive_tds_pdu() [with Implementation = tdsl::net::tdsl_netimpl_arduino<WiFiClient>; uint32_t = unsigned int] network_io_base.h` which I have no idea how to fix it

**Reply by xmkg** [reply]
Hi @kangyite, can you share a minimal reproducer sketch?

**Reply by kangyite** [reply]
I am unable to reproduce it consistently. I think it happened when my wifi is bad
